src/Middleware/UnauthorizedHandler/CakeRedirectHandler.php

```diff
@@ -77,6 +77,11 @@ protected function getUrl(ServerRequestInterface $request, array $options): stri
             $url['?'][$options['queryParam']] = $redirect;
         }
 
+        $requestBase = $request->getAttribute('base');
+        if ($requestBase) {
+            $url['_base'] = $requestBase;
+        }
+
         return Router::url($url);
```

Comment:
Shouldn't `Router::url()` here handle the base path? The test might just need to use `Router::setRequest()`?